src/vivarium/examples/disease_model/risk.py

```diff
@@ -130,26 +127,32 @@ def setup(self, builder: Builder) -> None:
             f"{self.disease_rate}.population_attributable_fraction",
             modifier=self.population_attributable_fraction,
             component=self,
-            required_resources=[self.base_risk_exposure, self.relative_risk_pipeline],
+            required_resources=[self.base_exposure_pipeline, self.relative_risk_pipeline],
         )
         builder.value.register_attribute_modifier(
             f"{self.disease_rate}",
             modifier=self.rate_adjustment,
             component=self,
-            required_resources=[self.actual_risk_exposure, self.relative_risk_pipeline],
+            required_resources=[self.exposure_pipeline, self.relative_risk_pipeline],
         )
 
     ##################################
     # Pipeline sources and modifiers #
     ##################################
 
-    def population_attributable_fraction(self, index):
-        exposure = self.base_risk_exposure(index)
-        relative_risk = self.population_view.get_attributes(index, self.relative_risk_pipeline)
+    def population_attributable_fraction(self, index: pd.Index[int]) -> pd.Series[float]:
+        pop = self.population_view.get_attributes(
+            index, [self.base_exposure_pipeline, self.relative_risk_pipeline]
+        )
+        exposure = pop[self.base_exposure_pipeline]
+        relative_risk = pop[self.relative_risk_pipeline]
         return exposure * (relative_risk - 1) / (exposure * (relative_risk - 1) + 1)
 
-    def rate_adjustment(self, index, rates):
-        exposed = self.actual_risk_exposure(index)
-        rr = self.population_view.get_attributes(index, self.relative_risk_pipeline)
+    def rate_adjustment(self, index: pd.Index[int], rates: pd.Series[float]) -> pd.Series[float]:
+        pop = self.population_view.get_attributes(
+            index, [self.exposure_pipeline, self.relative_risk_pipeline]
+        )
+        exposed = pop[self.exposure_pipeline]
```

Comment:
You might be able to save three lines if you just call `get_attributes()` separately for each column